packages/flutter_tools/lib/src/base/analyze_size.dart

```diff
@@ -0,0 +1,368 @@
+        }
+        childWithPathAsName.addSize(pathsToSize[paths]);
+        currentNode = childWithPathAsName;
+      }
+      currentNode = rootNode;
+    }
+
+    return rootNode;
+  }
+
+  /// Prints all children paths for the lib/ directory in an APK.
+  /// 
+  /// A brief summary of aot snapshot is printed under 'lib/arm64-v8a/libapp.so'.
+  void _printLibChildrenPaths(
+    _SymbolNode currentNode,
+    String totalPath,
+    _SymbolNode aotSnapshotJsonRoot,
+  ) {
+    totalPath += currentNode.name;
+
+    if (currentNode.children.isNotEmpty && !currentNode.name.contains('libapp.so')) {
+      for (final _SymbolNode child in currentNode.children) {
+        _printLibChildrenPaths(child, '$totalPath/', aotSnapshotJsonRoot);
+      }
+    } else {
+      // Print total path and size if currentNode does not have any chilren.
+      _printEntitySize(totalPath, byteSize: currentNode.byteSize, level: 2);
+
+      // We picked this file because arm64-v8a is likely the most popular
+      // architecture. ther architecture sizes should be similar.
+      const String libappPath = 'lib/arm64-v8a/libapp.so';
+      // TODO(peterdjlee): Analyze aot size for all platforms.
+      if (totalPath.contains(libappPath)) {
+        _printAotSnapshotSummary(aotSnapshotJsonRoot);
+      }
+    }
+  }
+
+  /// Go through the AOT gen snapshot size JSON and print out a collapsed summary
+  /// for the first package level.
+  void _printAotSnapshotSummary(_SymbolNode aotSnapshotRoot, {int maxDirectoriesShown = 10}) {
+    _printEntitySize(
+      'Dart AOT symbols accounted decompressed size',
+      byteSize: aotSnapshotRoot.byteSize,
+      level: 3,
+    );
+
+    final List<_SymbolNode> sortedSymbols = aotSnapshotRoot.children.toList()
+      ..sort((_SymbolNode a, _SymbolNode b) => b.byteSize.compareTo(a.byteSize));
+    for (final _SymbolNode node in sortedSymbols.take(maxDirectoriesShown)) {
+      _printEntitySize(node.name, byteSize: node.byteSize, level: 4);
+    }
+  }
+
+  /// Adds breakdown of aot snapshot data as the children of the node at the given path.
+  Map<String, dynamic> _addAotSnapshotDataToApkAnalysis({
+    @required Map<String, dynamic> apkAnalysisJson,
+    @required List<String> path,
+    @required Map<String, dynamic> aotSnapshotJson,
+  }) {
+    Map<String, dynamic> currentLevel = apkAnalysisJson;
+    while (path.isNotEmpty) {
+      final List<Map<String, dynamic>> children = currentLevel['children'] as List<Map<String, dynamic>>;
+      final Map<String, dynamic> childWithPathAsName = children.firstWhere(
+        (Map<String, dynamic> child) => child['n'] as String == path.first,
+      );
+      path.removeAt(0);
+      currentLevel = childWithPathAsName;
+    }
+    currentLevel['children'] = aotSnapshotJson['children'];
+    return apkAnalysisJson;
+  }
+
+  /// A pretty printer for an entity with a size.
+  void _printEntitySize(
+    String entityName, {
+    @required int byteSize,
```

Comment:
formatting is off a bit here, indent the required params by 2 spaces

Reply:
Done